[IR] Sent decoded values back to the controller

jimmys01 opened the pull request.

jimmys01 committed:
[IR] Sent decoded values back to the controller

jimmys01 commented:
this breaks the rules by the way

jimmys01 commented:
this is mostly for testing for whomever is interested

jimmys01 committed:
Comment out interfering code

jimmys01 committed:
Force rebuild

jimmys01 commented:
> The job exceeded the maximum time limit for jobs, and has been terminated

TD-er commented:
What to do with this PR?
It does break rules like you stated and it is using an older version of the library.

jimmys01 commented:
I will resolve its conflicts, now about the rules thing I do not know how to tackle it. #2615 has some recommendations.

In my opinion we have to send the decimal representation of the signal to the rules engine and the string representation at the controller

TD-er commented:
Keep in mind the String type is currently highly experimental and there is only one plugin using it (Serial Proxy).
This one is also still in development.

So I'm not sure if it is wise to actually having this plugin also using the String type, since the IR plugin is actually being used by people and not in a test setup like the Serial Proxy.

jimmys01 committed:
Merge remote-tracking branch 'upstream/mega' into RX_pass_string_val

jimmys01 committed:
Resolve code conflicts

jimmys01 committed:
Merge remote-tracking branch 'upstream/mega' into RX_pass_string_val

jimmys01 committed:
[IR RX] Add the extra ' needed for the parsing of the JSON command

jimmys01 committed:
[IR RX] Option to Send IR events to MQTT broker

Set a compile time option to send the String representation of the IR signal back to the conrroller.
Defaults to the old behaviour so it does not brake anything.

jimmys01 commented:
It can safely merge now

jimmys01 commented:
@TD-er are you going to merge this?

TD-er commented:
> @TD-er are you going to merge this?

Done.
Kinda forgot about this one.
